lib/ash_admin/resource/transformers/add_position_sort_calculation.ex

```diff
@@ -0,0 +1,91 @@
+defmodule AshAdmin.Resource.Transformers.AddPositionSortCalculation do
+  @moduledoc """
+  Adds a `ash_admin_position_sort` calculation to resources that have defined admin.label_field.
+  This calculation is used when loading typeahead search suggestions in a related resource field.
+  """
+
+  use Spark.Dsl.Transformer
+  use Ash.Resource.Calculation
+
+  require Logger
+
+  alias AshAdmin.Resource.Transformers.AddPositionSortCalculation, as: AddPositionSortCalculation
+  alias Spark.Dsl.Transformer
+
+  @impl true
+  def transform(dsl) do
+    case Transformer.get_persisted(dsl, :data_layer) do
+      nil ->
+        {:ok, dsl}
+
+      data_layer ->
+        case Transformer.get_option(dsl, [:admin], :label_field) do
+          nil ->
+            {:ok, dsl}
+
+          label_field ->
+            opts = [
+              name: :ash_admin_position_sort,
+              type: :integer,
+              calculation: calculation(label_field, data_layer),
+              arguments: [
+                %{name: :search_term, type: :string, constraints: [], default: ""}
+              ],
+              sortable?: true,
+              load: [field: label_field]
+            ]
+
+            case Transformer.build_entity(
+                   Ash.Resource.Dsl,
+                   [:calculations],
+                   :calculate,
+                   opts
+                 ) do
+              {:ok, calculation} ->
+                {:ok, Transformer.add_entity(dsl, [:calculations], calculation)}
+
+              error ->
+                error
+            end
+        end
+    end
+  end
+
+  defp calculation(label_field, AshPostgres.DataLayer) do
+    expr(
+      fragment(
```

Comment:
I think we can use `contains(^ref(label_field), ^search_term)` instead of this I think.

Reply:
If not, I'd like to add a `starts_with` expression to `core` to make this work 😄

Reply:
The calculation is used to rank results based on the position of the search term in the matched result. I kinda thought it made for neat UI, in case all the user can do is guess at keywords in a large data set, but maybe that's overkill.

If I'm understanding your suggestion correctly, and if you do prefer returning only results that _start with_ the user supplied string, then I can go ahead and make that change.

Reply:
I also don't mind contributing `starts_with` as you suggested, regardless of the decision for ash_admin.

Reply:
Right right, we should add `string_position` instead of `starts_with` so you can rank them the way you have here 😄